src/components/Login.jsx

```diff
@@ -1,37 +1,64 @@
-import React from 'react';
+import { useState } from 'react';
 import { Form, Button, Card } from 'react-bootstrap';
-import '../stylesheets/Login.css'
+import '../stylesheets/Login.css';
+import UserService from '@services/userService';
+import { useNavigate } from 'react-router';
 
 const Login = () => {
+    const navigate = useNavigate();
+    const [credentials, setCredentials] = useState({ email: '', password: '' });
+    const handleChange = (e) => {
+        setCredentials({
+            ...credentials,
+            [e.target.name]: e.target.value
+        })
+    }
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try {
+            const response = await UserService.login(credentials);
+            const token = response.token;
+            localStorage.setItem('token', token);
+            navigate('/me');
+        } catch (error) {
+            console.error('Login failed', error);
+        }
+    }
     return (
         <div className='div-login'>
             <Card className="login-card shadow-lg">
-            <Card.Body>
-                <h2 className="text-center mb-4">Login</h2>
-                <Form>
-                    <Form.Group className="mb-3" controlId="formBasicEmail">
-                        <Form.Control type="email" placeholder="Email" />
-                    </Form.Group>
+                <Card.Body>
+                    <h2 className="text-center mb-4">Login</h2>
+                    <Form onSubmit={handleSubmit}>
+                        <Form.Group className="mb-3" controlId="formBasicEmail">
+                            <Form.Control
+                                type="email"
+                                name="email"
+                                placeholder="Email"
+                                value={credentials.email}
+                                onChange={handleChange}
+                            />
+                        </Form.Group>
 
-                    <Form.Group className="mb-3" controlId="formBasicPassword">
-                        <Form.Control type="password" placeholder="Password" />
-                    </Form.Group>
+                        <Form.Group className="mb-3" controlId="formBasicPassword">
+                            <Form.Control type="password" name="password" placeholder="Password" value={credentials.password} onChange={handleChange} />
+                        </Form.Group>
 
-                    <div className="d-grid">
-                        <Button variant="primary" type="submit" className="gradient-btn">
-                            Login
-                        </Button>
-                    </div>
-                </Form>
+                        <div className="d-grid">
+                            <Button variant="primary" type="submit" className="gradient-btn">
```

Comment:
Let’s disable the submit button if the username or password isn’t set.

Also, please add a new validation: the password must be at least 8 characters long and include at least one uppercase letter, one lowercase letter, and one special character (e.g., comma, colon, parentheses, etc.).